pkg/ruler/notifier.go

```diff
@@ -74,18 +72,24 @@ func (rn *rulerNotifier) stop() {
 // Builds a Prometheus config.Config from a ruler.Config with just the required
 // options to configure notifications to Alertmanager.
 func buildNotifierConfig(rulerConfig *Config) (*config.Config, error) {
-	if rulerConfig.AlertmanagerURL.URL == nil {
+	validHosts := make([]string, 0, len(rulerConfig.AlertmanagerHosts))
+
+	for _, h := range rulerConfig.AlertmanagerHosts {
+		if h == "" {
+			continue
+		}
+
+		validHosts = append(validHosts, h)
+	}
+
+	if len(validHosts) == 0 {
 		return &config.Config{}, nil
 	}
 
-	u := rulerConfig.AlertmanagerURL
 	var sdConfig sd_config.ServiceDiscoveryConfig
 	if rulerConfig.AlertmanagerDiscovery {
-		if !strings.Contains(u.Host, "_tcp.") {
```

Comment:
I can't seem to find a history behind this change and the reason for its introduction but my investigation points out that this is not needed e.g. using a Kubernetes headless service both `_http._tcp.service.svc.local` and `service.svc.local` produce the same DNS query result.

Reply:
Cleared offline with @gouthamve, I have reintroduced the check given we need it. Added a test for it and a comment that explains a bit more in depth the reason.